src/SessionSetup/TargetLabel.cpp

```diff
@@ -88,6 +88,8 @@ void TargetLabel::ChangeToFileTarget(const FileTarget& file_target) {
 }
 
 void TargetLabel::SetFile(const std::filesystem::path& file_path) {
+  if (file_path_ == file_path) return;
```

Comment:
maybe add a comment here?

Reply:
Good idea, I've updated the commit to explain why this is needed, thanks!